proposals/NNNN-atf-script-runner.md

```diff
@@ -0,0 +1,43 @@
+# ATF Script Runner
+
+* Proposal: [SDL-NNNN](NNNN-atf-script-runner.md)
+* Author: [Dmytro Boltovskyi](https://github.com/dboltovskyi)
+* Status: **Awaiting review**
+* Impacted Platforms: [ATF]
+
+## Introduction
+
+Originally ATF was developed for SDL developers and thats why it lacks a few features of modern Test Frameworks
+(Java jUnit, Python Nose, etc.)
+
+The purpose of this proposal is to develop extended script runner as part of ATF functionality.
+
+## Motivation
+
+Currently ATF is unable to:
+
+  - run multiple test scripts
+  - run test script in parallel
+  - create test reports in standard format
+
+## Proposed solution
+
+Develop script runner with the following features:
+
+  - Ability to run single test script
+  - Ability to run batch of test scripts
+  - Ability to run test set
+  - Possibility to run scripts in parallel
```

Comment:
@dboltovskyi 
Is that really possilble w/o side effects? I mean if several test scripts will be run simultaneously and use same SDL instance they may have affect on each other. In case there will be several SDL instances again we may face with same problem due to connection parameters, communication channels simultaneous usage etc.

Reply:
@dev-gh Yep - with the same instance of SDL it's impossible. But the idea is to use SDL Watchdog (new feature in separate proposal) that will provide a new SDL instance for test script from the pull of available instances.